ts/Stock/StockToolsGui.ts

```diff
@@ -1620,6 +1620,11 @@ class Toolbar {
         if (this.chart.navigationBindings) {
             this.chart.navigationBindings.update();
         }
+
+        if (pick(redraw, true)) {
+            this.chart.isDirtyBox = true;
```

Comment:
The idea of the dirty/redraw design is that we mark items dirty when updated, but the redraw itself is updated. So that we can optimize chart updates by doing only one chart redraw. Consequently, `isDirtyBox` must be set regardless of `redraw` (move it outside the condition).